toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Enum.jsm

```diff
@@ -0,0 +1,103 @@
+{%- let enum_ = ci.get_enum_definition(name).unwrap() %}
+
+{%- if enum_.is_flat() %}
+
+const {{ enum_.nm() }} = {
+    {%- for variant in enum_.variants() %}
+    {{ variant.name().to_shouty_snake_case() }}: {{loop.index}},
```

Comment:
I think I slightly prefer having an `Enum.variant_name(&Variant)` function for this.  That would also be useful for literal rendering.  This is minor though, and it would be fine to figure this out down the road.

Reply:
Yeah, that might require us making `Variant` public 😢 I would like to come back to this, and other minor clean ups after